### PR TITLE
feat: Person detail related feature flags

### PR DESCRIPTION
**Pull request by liyiy**
## Changes

Based off of these mocks: [figma link](https://www.figma.com/file/XklcUpnAb8UYXJxVSZFNeI/Cohorts%2C-Persons-%26-Groups?node-id=2%3A2)

Screen Shot 2022-09-07 at 5 19 19 PM

(purple experiment tags)
Screen Shot 2022-09-07 at 2 30 02 PM

- video of filtering

https://user-images.githubusercontent.com/25164963/188968014-85a90731-42fd-4b3a-950c-f6bdae42e5b4.mov

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: frontend/src/scenes/persons/RelatedFeatureFlags.tsx [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+export enum FeatureFlagMatchReason {
+    ConditionMatch = 'condition_match',
+    NoConditionMatch = 'no_condition_match',
+    OutOfRolloutBound = 'out_of_rollout_bound',
+    NoGroupType = 'no_group_type',
+    Disabled = 'disabled',
+}
+
+const featureFlagMatchMapping = {
+    [FeatureFlagMatchReason.ConditionMatch]: 'Matches',
```
Comment (liyiy) [inline frontend/src/scenes/persons/RelatedFeatureFlags.tsx:25]: @neilkakkar @clarkus We're only returned that it "matches" and not which filters it's matching from the API, which is why I've left it as "matches"

Comment (liyiy) [reply frontend/src/scenes/persons/RelatedFeatureFlags.tsx:25]: Not sure if there are limiting reasons as to why the API doesn't return which set of filters the person belongs to @neilkakkar

Comment (neilkakkar) [reply frontend/src/scenes/persons/RelatedFeatureFlags.tsx:25]: The API does return this! It's the `condition_index` part of `evaluation`.

Comment (neilkakkar) [reply frontend/src/scenes/persons/RelatedFeatureFlags.tsx:25]: If condition_index is set, it refers to zero-indexed group that it matched.

Comment (liyiy) [reply frontend/src/scenes/persons/RelatedFeatureFlags.tsx:25]: hahahah ok then

Comment (liyiy) [reply frontend/src/scenes/persons/RelatedFeatureFlags.tsx:25]: done

Screen Shot 2022-09-07 at 5 17 05 PM

File: frontend/src/scenes/persons/Person.tsx [hunk 2]
```diff
@@ -207,6 +208,14 @@ export function Person(): JSX.Element | null {
                         <RelatedGroups id={person.uuid} groupTypeIndex={null} />
                     </TabPane>
                 )}
+                {person.uuid && (
+                    <TabPane
+                        tab={<span data-attr="persons-related-flags-tab">Feature flags</span>}
+                        key={PersonsTabType.FEATURE_FLAGS}
+                    >
+                        <RelatedFeatureFlags distinctId={person.uuid} />
```
Comment (neilkakkar) [inline frontend/src/scenes/persons/Person.tsx:216]: this needs to be a distinctID, not the uuid. Giving wrong results right now 😅

File: frontend/src/scenes/persons/RelatedFeatureFlags.tsx [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+                        value="all"
+                        dropdownMaxContentWidth
+                    />
+                    <span>
+                        <b>Match Evaluation</b>
+                    </span>
+                    <LemonSelect
+                        options={[
+                            { label: 'All', value: 'all' },
+                            ...Object.values(FeatureFlagMatchReason).map((value) => ({
```
Comment (neilkakkar) [inline frontend/src/scenes/persons/RelatedFeatureFlags.tsx:149]: mocks seem to suggest this should be `All`, `match`, or `no match`

File: frontend/src/scenes/persons/relatedFeatureFlagsLogic.ts [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+import { actions, connect, events, kea, key, path, props, selectors, reducers } from 'kea'
+import { loaders } from 'kea-loaders'
+import api from 'lib/api'
+import { toParams } from 'lib/utils'
+import { featureFlagsLogic } from 'scenes/feature-flags/featureFlagsLogic'
+import { teamLogic } from 'scenes/teamLogic'
+import { FeatureFlagReleaseType, FeatureFlagType } from '~/types'
+
+import type { relatedFeatureFlagsLogicType } from './relatedFeatureFlagsLogicType'
+export interface RelatedFeatureFlag extends FeatureFlagType {
```
Comment (neilkakkar) [inline frontend/src/scenes/persons/relatedFeatureFlagsLogic.ts:11]: this seems unused?

File: frontend/src/scenes/persons/Person.tsx (hunk 2 above)
Comment (liyiy) [reply frontend/src/scenes/persons/Person.tsx:216]: distinctId comes in as an array, do you want me to just grab the first one from the array or?

File: frontend/src/scenes/persons/RelatedFeatureFlags.tsx (hunk 3 above)
Comment (liyiy) [reply frontend/src/scenes/persons/RelatedFeatureFlags.tsx]: ✅

File: frontend/src/scenes/persons/Person.tsx (hunk 2 above)
Comment (liyiy) [reply frontend/src/scenes/persons/Person.tsx]: Added it as the first option from the array, can change when you're back from haircut if it's wrong 😛

**Comment by liyiy** [comment]
Most recent updates:

Screen Shot 2022-09-08 at 11 34 55 AM

File: frontend/src/scenes/persons/Person.tsx (hunk 2 above)
Comment (neilkakkar) [reply frontend/src/scenes/persons/Person.tsx]: lol, yeah, good enough!

**Review by clarkus — CHANGES_REQUESTED** [review]
The filters at the top of the table could use a bit more space between options. In the current layout the label for `match evaluation` is just a close to its corresponding input as it is to the previous `type` input. A bit more space between options would better associate the labels with their corresponding fields. Typically labels are 8px from their corresponding inputs, while the spacing between label/input pairs is 16px.

**Comment by liyiy** [comment]
> The filters at the top of the table could use a bit more space between options. In the current layout the label for `match evaluation` is just a close to its corresponding input as it is to the previous `type` input. A bit more space between options would better associate the labels with their corresponding fields. Typically labels are 8px from their corresponding inputs, while the spacing between label/input pairs is 16px.

Spacing should be 16px apart now to the left of the labels, let me know if it needs more !

Screen Shot 2022-09-08 at 5 38 54 PM

**Comment by clarkus** [comment]
Spacing looks better but now that match evaluation label is all caps. It should be the same font style as the other labels.

**Comment by liyiy** [comment]
> Spacing looks better but now that match evaluation label is all caps. It should be the same font style as the other labels.

I tried to match it to the figma mocks after updating the padding 😞 but is no biggie to switch back 😃 

-- mocks
Screen Shot 2022-09-08 at 11 51 24 PM

-- back to lowercase now

Screen Shot 2022-09-09 at 11 22 29 AM

**Comment by clarkus** [comment]
My mistake! I've fixed it in [figma](https://www.figma.com/file/XklcUpnAb8UYXJxVSZFNeI/Cohorts%2C-Persons-%26-Groups?node-id=5%3A2029) and it aligns to what you have here.

**Review by clarkus — APPROVED** [review]

File: frontend/src/lib/components/LemonTag/LemonTag.scss [hunk 5]
```diff
@@ -30,11 +29,17 @@
         color: var(--light);
     }
 
+    &.purple {
+        background-color: #ebe3ed;
+        color: var(--indigo);
+    }
```
Comment (Twixes) [inline frontend/src/lib/components/LemonTag/LemonTag.scss:35]: Hmm, `purple` doesn't seem to fit in conceptually – it's a tag type, but the existing types are `'highlight' | 'warning' | 'danger' | 'success' | 'default'`. None of them is a color.

Also, this introduces a color that's not defined as a variable in `root-variables` – this is dangerous as we can easily introduce dozens of slightly different colors variations this way, which is hard to keep track of! Pretty much all colors should be defined at the top level.

File: frontend/src/lib/components/LemonTag/LemonTag.scss [hunk 6]
```diff
@@ -4,7 +4,6 @@
     background: var(--border);
     padding: 0.125rem 0.25rem;
     border-radius: var(--radius);
-    text-transform: uppercase;
```
Comment (Twixes) [inline frontend/src/lib/components/LemonTag/LemonTag.scss:7]: This broke the pre-existing 30 instances of `LemonTag`! They were all intended to be all-caps and now accidentally aren't. It's totally okay to change the styling of a component, but existing UI needs to be taken into account.

**Review by Twixes — COMMENTED** [review]
Post-mortem as I was looking into a UI regression

**Pull request merged**